doc/specs/stdlib_linalg.md

```diff
@@ -168,3 +168,41 @@ program demo_trace
     print *, trace(A) ! 1 + 5 + 9
 end program demo_trace
 ```
+
+## `outer_product` - Computes the outer product of two vectors
+
+### Status
+
+Experimental
+
+### Description
+
+Create a diagonal array or extract the diagonal elements of an array
+
+### Syntax
+
+`d = [[stdlib_linalg(module):outer_product(interface)]](u, v)`
+
+### Arguments
+
+`u`: Shall be a rank-1 array
+
+`v`: Shall be a rank-1 array
+
+### Return value
+
+Returns a rank-2 array equal to `u v^T` (assuming `u, v` are column vectors). The dimensions of the returned array are `[size(u), size(v)]`
+
+### Example
+
+```fortran
+program demo_outer_product
+    use stdlib_linalg, only: outer_product
+    implicit none
+    real, allocatable :: A(:,:), u(:), v(:)
+    u = [1.,2.,3.]
+    v = [3., 4.]
+    A=outer_product(u,v)
+    !A = transpose(reshape([3., 4., 6., 8., 9., 12.], [3,2]))
```

Comment:
The transpose of a 3×2 array will be a 2×3 array, which disagrees with the description of `outer_product`. I think it's best to just use a reshape, and specify column by column:

 ```
! A = reshape([3., 6., 9., 4., 8., 12.], [3, 2])
```